cpp/src/strings/copying/concatenate.cu

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <bitmask/legacy/valid_if.cuh>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/strings/detail/concatenate.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+#include "../utilities.hpp"
+
+#include <thrust/for_each.h>
+#include <thrust/transform_reduce.h>
+
+namespace cudf
+{
+namespace strings
+{
+namespace detail
+{
+
+std::unique_ptr<column> concatenate_vertically( std::vector<strings_column_view> const& strings_columns,
+                                                rmm::mr::device_memory_resource* mr,
+                                                cudaStream_t stream )
+{
+    if( strings_columns.size() == 0 )
```

Comment:
I think you can eliminate this one; I expect the transform_reduce will return 0 quite quickly and the result is the same.